contrib/xor-blocks/src/main.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    path::Path,
+    time::{Duration, Instant},
+};
+
+const MAGIC: [u8; 4] = [0xf9, 0xbe, 0xb4, 0xd9];
+
+fn main() -> Result<(), io::Error> {
+    let start = Instant::now();
+
+    let blocks_path = if env::args().len() > 1 {
+        env::args().nth(1).expect("the arg to exist").into()
+    } else {
+        #[allow(deprecated)]
+        env::home_dir()
+            .expect("to have a home directory")
+            .join(match env::consts::OS {
+                "macos" => "Library/Application Support/Bitcoin",
+                "windows" => "AppData\\Local\\Bitcoin",
+                "linux" => ".bitcoin",
+                _ => {
+                    println!("Unknown OS");
+                    return Ok(());
+                }
+            })
+            .join("blocks")
+    };
+
+    let paths = fs::read_dir(&blocks_path)?
+        .map(|res| res.map(|e| e.path()))
+        .collect::<Result<Vec<_>, io::Error>>()?;
+
+    let xor_path: std::path::PathBuf = blocks_path.join("xor.dat");
+    if !fs::exists(&xor_path)? {
+        println!("No xor.dat file. Make sure you are running Bitcoin Core v28 or higher.");
+        return Ok(());
+    }
+
+    println!("Xor'ing blocks dir. Do not start bitcoind until finished!");
```

Comment:
xor is an implementation detail - I'd make it more user friendly and call it obfuscation instead

Reply:
Replaced all logs of xor with obfuscation.